wordpress-mobile/WordPress-iOS
# Update screenshot test with new mocks

Updates the screenshot test with new mock data for new app store screenshots.

### Notable changes:

* Depends on https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/493 to support running the tests in any language.
* Depends on https://github.com/wordpress-mobile/WordPressMocks/pull/20 for the new mocks.
* Adds a new accessibility ID to the comment notification reply button.
* Makes some updates to the screenshot test to support the new mocked data (e.g. switching sites).
* Makes some updates to the UI tests and screen objects to support the new mocked data.

Note: We're aiming to use the new screenshots with the 15.9 release, but I'm targeting `develop` because we don't expect any significant UI differences between the `release/15.9` and `develop` branches.

#### Screenshots

Sample iPad screenshots:

Editor|Reader|My Site/Stats|Notifications|Editor with keyboard|Media
--|--|--|--|--|--
<img width="1366" alt="iPad Pro (12 9-inch) (2nd generation)-1-Editor-With-BlockPicker" src="https://user-images.githubusercontent.com/8658164/95195829-ff21e180-07ce-11eb-8e60-8a5b32d34464.png">|<img width="1366" alt="iPad Pro (12 9-inch) (2nd generation)-2-Discover" src="https://user-images.githubusercontent.com/8658164/95195845-03e69580-07cf-11eb-9207-3e78bd15d12d.png">|<img width="1366" alt="iPad Pro (12 9-inch) (2nd generation)-3-Stats" src="https://user-images.githubusercontent.com/8658164/95195847-047f2c00-07cf-11eb-9231-99a1ec37d9c3.png">|<img width="1366" alt="iPad Pro (12 9-inch) (2nd generation)-5-Notifications" src="https://user-images.githubusercontent.com/8658164/95195849-0517c280-07cf-11eb-8294-d5fafef35fa2.png">|<img width="1366" alt="iPad Pro (12 9-inch) (2nd generation)-6-Editor-With-Keyboard" src="https://user-images.githubusercontent.com/8658164/95195852-05b05900-07cf-11eb-8a7f-0a31e6c4a734.png">|<img width="1366" alt="iPad Pro (12 9-inch) (2nd generation)-6-Media" src="https://user-images.githubusercontent.com/8658164/95195857-06e18600-07cf-11eb-99a6-c97b31ed260b.png">


Sample iPhone screenshots:

Editor|Reader|Stats|My Site|Notifications|Media
--|--|--|--|--|--
![iPhone Xs Max-1-Editor](https://user-images.githubusercontent.com/8658164/95195527-93d80f80-07ce-11eb-9cea-d699b0cb05b6.png)|![iPhone Xs Max-2-Discover](https://user-images.githubusercontent.com/8658164/95195537-98042d00-07ce-11eb-807a-d466198a1f5d.png)|![iPhone Xs Max-3-Stats](https://user-images.githubusercontent.com/8658164/95195540-99cdf080-07ce-11eb-9ca2-4272d65a63f2.png)|![iPhone Xs Max-4-MySite](https://user-images.githubusercontent.com/8658164/95195541-99cdf080-07ce-11eb-8e21-6aaf432a8e56.png)|![iPhone Xs Max-5-Notifications](https://user-images.githubusercontent.com/8658164/95195543-9a668700-07ce-11eb-9aab-7d9fababb9e4.png)|![iPhone Xs Max-6-Media](https://user-images.githubusercontent.com/8658164/95195544-9aff1d80-07ce-11eb-8baa-99e8c64c6ba2.png)

### To test:

* Confirm all the regular UI tests pass (WordPressUITests) on this PR.
* Confirm the WordPressScreenshotGeneration tests produces the expected screenshots:
  * Run `rake mocks` to start the mock server.
  * Open the fastlane directory with `cd Scripts/fastlane`.
  * Generate a set of screenshots: `bundle exec fastlane screenshots` for a full set in all languages (this takes a while) or e.g. `bundle exec fastlane screenshots language:"es-ES"` for a set of screenshots in a single non-English language to confirm it works for all targeted devices regardless of language.

### Before merge

- [x] Update the Podfile/Podfile.lock to use to the published WordPressAuthenticator and WordPressMocks pods (once those PRs are merged).

### PR submission checklist:
